src/script/api/script_goal.cpp

```diff
@@ -117,10 +117,12 @@
 	EnforcePrecondition(false, buttons < (1 << ::GOAL_QUESTION_BUTTON_COUNT));
 	EnforcePrecondition(false, (int)type < ::GQT_END);
 
+	uniqueid = Clamp<SQInteger>(uniqueid, 0, UINT16_MAX);
```

Comment:
With IDs I'm wondering whether clamping is the right solution. Now every negative number becomes 0, so a lot of questions could be asked with the same ID. Maybe for IDs there should be an `EnforcePrecondition` of sorts that checks the range.